llvm/include/llvm/SandboxIR/SandboxIR.h

```diff
@@ -71,6 +71,98 @@ namespace sandboxir {
 class Function;
 class Context;
 class Instruction;
+class User;
+class Value;
+
+/// Represents a Def-use/Use-def edge in SandboxIR.
+/// NOTE: Unlike llvm::Use, this is not an integral part of the use-def chains.
+/// It is also not uniqued and is currently passed by value, so you can have
+/// more than one sandboxir::Use objects for the same use-def edge.
+class Use {
+  llvm::Use *LLVMUse;
+  User *User;
+  Context *Ctx;
+
+  /// Don't allow the user to create a sandboxir::Use directly.
+  Use(llvm::Use *LLVMUse, class User *User, Context &Ctx)
+      : LLVMUse(LLVMUse), User(User), Ctx(&Ctx) {}
+  Use() : LLVMUse(nullptr), Ctx(nullptr) {}
+
+  friend class Value;              // For constructor
+  friend class User;               // For constructor
+  friend class OperandUseIterator; // For constructor
+  friend class UserUseIterator;    // For accessing members
+
+public:
+  operator Value *() const { return get(); }
```

Comment:
implicit conversions aren't great, do we need this (and all the other ones)?

Reply:
I did this because this is how it's done in LLVM IR's `Use` class. This is used for example by `getOperand(Idx)` which internally finds the Use at `Idx` and casts it to `Value *` to return it.
Not sure how often this is used in practice by passes, so I don't feel strongly about keeping it. Wdyt?

Reply:
alright let's mimic the llvm API for now